src/armory/CopyCasusCodePopup.js

```diff
@@ -0,0 +1,95 @@
+//@flow strict
+
+import * as React from 'react';
+import Popup from 'reactjs-popup';
+import Tank from '../tanks/Tank.js';
+import { ToastContainer , toast } from 'react-toastify';
+import { updateCasusCode } from '../globalComponents/apiCalls/tankAPIIntegration';
+
+type Props = {|
+	selectedTank: Tank,
+	usersTanks: Array<Tank>,
+|};
+
+type State = {|
+	popupOpen: boolean,
+	tankBeingCopiedFrom: ?Tank,
+|};
+
+class CopyCasusCodePopup extends React.Component<Props, State> {
+	
+	constructor(props: Props) {
+		super(props);
+
+		this.state = {
+			popupOpen: false,
+			tankBeingCopiedFrom : null,
+		}
+	}
+
+	setTankBeingCopiedFrom(tankBeingUsedId: string): void {
+		let indexOfTankBeingUsed = -1;
+		for(let i = 0; i < this.props.usersTanks.length; i++) {
+			if(this.props.usersTanks[i]._id === tankBeingUsedId) {
+				indexOfTankBeingUsed = i;
+				i = this.props.usersTanks.length;
+			}
+		}
+		if(indexOfTankBeingUsed === -1) {
+			this.setState({tankBeingCopiedFrom: null});
+			return;
+		}
+		this.setState({tankBeingCopiedFrom: this.props.usersTanks[indexOfTankBeingUsed]});
```

Comment:
If, for whatever reason, you can't just pass the tank object into this method, you can still clean it up quite a bit to something like:
`this.setState({tankBeingCopiedFrom: this.props.userTanks.find(tank => tank._id == tankBeingUsedId});`

Reply:
Obviously if passing in the tanks works, you can do just:
`this.setState({tankBeingCopiedFrom: tankBeingUsed});`